Python/Product/Analysis/Interpreter/Ast/NameLookupContext.cs

```diff
@@ -268,8 +265,9 @@ class NameLookupContext {
                         return new AstPythonConstant(Interpreter.GetBuiltinType(BuiltinTypeId.Bool), GetLoc(expr));
                 }
 
-                // When in doubt, assume that the type after the op is the same as the LHS
-                return GetValueFromExpression(binop.Left);
+                // Try LHS, then, if unknown, try RHS. Example: y = 1 when y is not declared by the walker yet.
```

Comment:
This isn't a great example, as `=` isn't a binary operator in Python. But it applies for `+` well enough